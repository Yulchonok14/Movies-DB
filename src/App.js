import React from 'react';

import Layout from './hoc/Layout/Layout';
import MovieHunter from './containers/MovieHunter/MovieHunter';
import MovieDetails from './containers/MovieDetails/MovieDetails';
import ErrorBoundary from './components/ErrorBoundary/ErrorBoundary';

const App = () => {
    return (
        <ErrorBoundary>
            <Layout>
                <MovieHunter/>
                <MovieDetails/>
            </Layout>
        </ErrorBoundary>
    )
};

export default App;