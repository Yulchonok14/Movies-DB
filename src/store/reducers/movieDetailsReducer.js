import * as actionTypes from './../actions/actionTypes';
import { sortFunc } from '../utils/storeUtils';

const initialState = {
    movieDetailsItems: [],
    chosenMovieItem: {
        budget: 0,
        genres: ["Action", "Adventure", "Science Fiction"],
        id: 447365,
        overview: "The third film based on Marvel's Guardians of the Galaxy.",
        poster_path: "https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg",
        release_date: "2020-05-01",
        revenue: 0,
        runtime: null,
        tagline: "",
        title: "Guardians of the Galaxy Vol. 3",
        vote_average: 0,
        vote_count: 9
    }
};

const movieDetailsReducer = (state = initialState, action) => {
    switch (action.type) {
        case actionTypes.GET_MOVIES_DETAILS:
            return {
                ...state,
                movieDetailsItems: action.payload.movieItems
                    .filter(item => item.id !== state.chosenMovieItem.id)
            };
        case actionTypes.CHOOSE_MOVIE_DETAILS:
            const chosenMovieDetails = [...state.movieDetailsItems].find(movie => movie.id === action.payload.id);
            return {
                ...state,
                chosenMovieItem: chosenMovieDetails
            };
        case actionTypes.PASS_CHOSEN_MOVIE:
            return {
                ...state,
                chosenMovieItem: action.payload.chosenMovie
            };
        default:
            return state
    }
};

export default movieDetailsReducer;