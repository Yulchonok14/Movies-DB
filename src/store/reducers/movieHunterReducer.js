import * as actionTypes from './../actions/actionTypes';
import { sortFunc } from '../utils/storeUtils';

const initialState = {
    movieHunterItems: [],
    searchFilter: 'title',
    sortFilter: 'release_date'
};

const movieHunterReducer = (state = initialState, action) => {
    switch (action.type) {
        case actionTypes.GET_MOVIES_HUNTER:
            return {
                ...state,
                movieHunterItems: action.payload.movieItems
                    .sort((a, b) => sortFunc(a, b, state.sortFilter))
            };
        case actionTypes.SEARCH_FILTER:
            return {
                ...state,
                searchFilter: action.payload.searchFilterText
            };
        case actionTypes.SORT_FILTER:
            return {
                ...state,
                sortFilter: action.payload.sortFilterText
            };
        case actionTypes.SORT_MOVIES:
            return {
                ...state,
                movieHunterItems: [...state.movieHunterItems]
                    .sort((a, b) => sortFunc(a, b, state.sortFilter))
            };
        default:
            return state
    }
};

export default movieHunterReducer;