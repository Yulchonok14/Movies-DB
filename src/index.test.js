import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import renderer from 'react-test-renderer';

//import { rootReducer } from './index';
import movieHunterReducer from './store/reducers/movieHunterReducer';
import movieDetailsReducer from './store/reducers/movieDetailsReducer';

jest.mock("react-dom", () => ({ render: jest.fn() }));

describe("Application root", () => {
    it('snapshot renders', () => {
        const store = createStore(() => {});
        const component = renderer.create(<Provider store={store}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    //it('rootReducer initial state matches result of child reducers with empty action', () => {
    //    let store = createStore(rootReducer);
    //    expect(store.getState().hunter).toEqual(movieHunterReducer(undefined, {}));
    //    expect(store.getState().details).toEqual(movieDetailsReducer(undefined, {}));
    //});
});