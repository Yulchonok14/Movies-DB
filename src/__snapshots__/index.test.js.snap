// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application root snapshot renders 1`] = `null`;
